timemachine/potentials/bonded_stable.py

```diff
@@ -0,0 +1,56 @@
+import jax.numpy as jnp
+
+
+def harmonic_angle_stable(conf, params, angle_idxs, cos_angles=True):
+    r"""
+    Compute the harmonic angle energy using a numerically stable approximation.
+
+    The functional form is identical to :py:func:`potentials.bonded.HarmonicAngle`, except that the following
+    approximation is used for the intermediate computation of :math:`\cos(\theta)`:
+
+    :math::
+
+        \cos(\theta) \approx \frac{r_{ij} \cdot r_{kj} + \epsilon^2}{\sqrt{(r_{ij}^2 + \epsilon^2) (r_{kj}^2 + \epsilon^2)}}
+
+    This reduces to the exact expression when :math:`\epsilon = 0`; When :math:`\epsilon > 0`, this avoids the
+    singularities in the exact expression as :math:`r_{ij}` or :math:`r_{kj}` approach zero.
```

Comment:
Docs tweak: May clarify that epsilons are used to stabilize both the cos_angles=True and cos_angles=False code paths?

Reply:
Done in eda6999